pytest_rpc/helpers.py

```diff
@@ -33,3 +39,187 @@ def get_osa_version_tuple():
+    """Create a bootable volume using a json file
+
+    Args:
+        data (dictionary): Dictionary in the following format:
+                           { 'volume': { 'size': '',
+                                         'imageref': '',
+                                         'name': '',
+                                         'zone': '',
+                                       }
+                           }
+        run_on_host (testinfra.Host): Testinfra host object to execute the
+                                      action on.
+    """
+
+    volume_size = data['volume']['size']
+    imageRef = data['volume']['imageRef']
+    volume_name = data['volume']['name']
+    zone = data['volume']['zone']
+
+    cmd = "{} openstack volume create \
+           --size {} \
+           --image {} \
+           --availability-zone {} \
+           --bootable {}'".format(utility_container, volume_size, imageRef,
+                                  zone, volume_name)
+    run_on_host.run_expect([0], cmd)
+
+
+def openstack_name_list(name, run_on_host):
+    """Get a list of OpenStack object instances of given type.
+
+    Args:
+        name (str): The OpenStack object type to query for.
+        run_on_host (testinfra.Host): Testinfra host object to execute the
+                                      action on.
+
+    Returns:
+        string: List of OpenStack object instances in table format.
+    """
+    cmd = "{} openstack {} list'".format(utility_container, name)
+    output = run_on_host.run(cmd)
+    return output.stdout
+
+
+def delete_volume(volume_name, run_on_host):
+    """Delete OpenStack volume
+
+    Args:
+        volume_name (str): OpenStack volume identifier (name or id).
+        run_on_host (testinfra.Host): Testinfra host object to execute the
+                                      action on.
+
+    Raises:
+        AssertionError: If operation unsucessful.
+    """
+    volume_id = get_id_by_name('volume', volume_name, run_on_host)
+    cmd = "{} openstack volume delete --purge {}'".format(utility_container,
+                                                          volume_id)
+    output = run_on_host.run(cmd)
+    assert volume_name not in output.stdout
+
+
+def parse_table(ascii_table):
+    """Parse an OpenStack ascii table
+
+    Args:
+        ascii_table (str): OpenStack ascii table.
+
+    Returns:
+        list: List of column headers from table.
+        list: List of column rows from table.
+    """
+    header = []
+    data = []
+    for line in filter(None, ascii_table.split('\n')):
+        if '-+-' in line:
+            continue
+        if not header:
+            header = filter(lambda x: x != '|', line.split())
+            continue
+        data.append([''] * len(header))
+        splitted_line = filter(lambda x: x != '|', line.split())
+        for i in range(len(splitted_line)):
+            data[-1][i] = splitted_line[i]
+    return header, data
+
+
+def generate_random_string(string_length=10):
```

Comment:
FYI: there is a maximum string length for this function given that UUID has only 32 alphanumeric characters.